src/SymfonyCache/TagSubscriber.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use FOS\HttpCache\Tag\ManagerInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use FOS\HttpCache\ProxyClient\Symfony;
+
+/**
+ * Adds tag invalidation capabilities to the Symfony HTTP cache
+ *
```

Comment:
we should explain the concept (header with tag information as json encoded, store cache items by tag so we can invalidate all items with that tag) a bit in this docblock